plugins/modules/ndo_physical_interface.py

```diff
@@ -0,0 +1,422 @@
+                match.details["interfaceDescriptions"] = updated_interface_descriptions
+            elif interface_descriptions == [] and match.details.get("interfaceDescriptions"):
+                ops.append(dict(op="remove", path="{0}/{1}/interfaceDescriptions".format(path, match.index)))
+
+            mso.sanitize(match.details)
+
+        else:
+            if not nodes:
+                mso.fail_json(msg=("ERROR: Missing 'nodes' for creating a Physical Interface."))
+
+            if not physical_interface_type:
+                mso.fail_json(msg=("ERROR: Missing physical interface type for creating a Physical Interface."))
+
+            payload = {
+                "name": physical_interface,
+                "templateId": mso_template.template.get("templateId"),
+                "schemaId": mso_template.template.get("schemaId"),
+                "nodes": nodes,
+                "interfaces": interfaces,
+                "policyGroupType": physical_interface_type,
+            }
+
+            if description:
+                payload["description"] = description
+
+            if physical_interface_type == "physical" and physical_policy_uuid:
+                payload["policy"] = physical_policy_uuid
+
+            if physical_interface_type == "breakout" and breakout_mode:
+                payload["breakoutMode"] = breakout_mode
+
+            if interface_descriptions:
+                payload["interfaceDescriptions"] = validate_interface_description(interface_descriptions)
+
+            ops.append(dict(op="add", path="{0}/-".format(path), value=copy.deepcopy(payload)))
+
+            mso.sanitize(payload)
+
+        mso.existing = mso.proposed
+
+    elif state == "absent":
+        if match:
+            ops.append(dict(op="remove", path="{0}/{1}".format(path, match.index)))
+
+    if not module.check_mode and ops:
+        response = mso.request(mso_template.template_path, method="PATCH", data=ops)
+        interface_policies = response.get("fabricResourceTemplate", {}).get("template", {}).get("interfaceProfiles", [])
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            interface_policies,
+            [KVPair("uuid", physical_interface_uuid) if physical_interface_uuid else KVPair("name", physical_interface)],
+        )
+        if match:
+            mso.existing = match.details
+        else:
+            mso.existing = {}
+    elif module.check_mode and state != "query":
+        mso.existing = mso.proposed if state == "present" else {}
+
+    mso.exit_json()
+
+
+def validate_interface_description(interface_descriptions):
+    interface_descriptions = [
+        {
+            "interfaceID": interface_description.get("interface_id"),
+            "description": interface_description.get("description"),
+        }
+        for interface_description in interface_descriptions
+    ]
+
+    return interface_descriptions
```

Comment:
Add it to module_utils/mso.py, use the one I made in https://github.com/CiscoDevNet/ansible-mso/pull/539/files
you can set `node` to an empty string.

Reply:
used